docs/components.rst

```diff
@@ -45,28 +45,41 @@ Well, in Discord, clicking buttons and using slash commands are called ``interac
 This method will return a :class:`ComponentContext <discord_slash.context.ComponentContext>` object that we can use to respond. For this example, we'll just edit the original message (:meth:`edit_origin() <discord_slash.context.ComponentContext.edit_origin>`)
 
 .. code-block:: python
 
     await ctx.send("My Message", components=[action_row])
     # note: this will only catch one button press, if you want more, put this in a loop
-    button_ctx: ComponentContext = await manage_components.wait_for_component(bot, action_row)
+    button_ctx: ComponentContext = await manage_components.wait_for_component(bot, components=action_row)
     await button_ctx.edit_origin(content="You pressed a button!")
 
 .. note:: It's worth being aware that if you handle the event in the command itself, it will not persist reboots. As such when you restart the bot, the interaction will fail
 
-Next we'll go over the alternative, a global event handler. This works just the same as :meth:`on_slash_command_error` or `on_ready`.
+Next we'll go over the alternative, a global event handler. This works just the same as :func:`on_slash_command_error` or `on_ready`. But note that this code will be triggered on any components interaction.
 
 .. code-block:: python
 
     @bot.event
     async def on_component(ctx: ComponentContext):
+        # you may want to filter or change behaviour based on custom_id or message
         await ctx.edit_origin(content="You pressed a button!")
 
+There is one more method - making a function that'll be component callback - triggered when components in specified messages or with specified custom_ids would be activated
+Let's register our callback function via decorator :meth:`component_callback() <discord_slash.client.SlashCommand.component_callback>`, in similar ways to slash commands.
+.. code-block:: python
+
+    @slash.component_callback()
+    async def hello(button_context: ComponentContext):
+        await ctx.edit_origin(content="You pressed a button!")
+
+In this example, :func:`hello` will be triggered when you receive interaction from component `custom_id` `"hello"`. Just like slash commands, `custom_id` of the callback defaults to the function name.
+You can also register such callbacks in cogs using :func:`cog_component() <discord_slash.cog_ext>`
+Additionally, component callbacks can be dynamically added (:meth:`add_component_callback() <discord_slash.client.SlashCommand.add_component_callback>`), removed (:meth:`remove_component_callback_obj() <discord_slash.client.SlashCommand.remove_component_callback_obj>`) or edited (:meth:`remove_component_callback() <discord_slash.client.SlashCommand.remove_component_callback>`, :meth:`extend_component_callback() <discord_slash.client.SlashCommand.extend_component_callback>`)
```

Comment:
Imo this just clutters the doc. I think you should just link to the relevant section of the docs, and let them read. No need to spam the docs with meth references

Reply:
In agreement with Polls, since there is information already written to this, we can refer to the existing documentation on that as support.